tests/interfaces/test_batch_torch.py

```diff
@@ -0,0 +1,947 @@
+
+        def cost(x, cache):
+            with qml.tape.JacobianTape() as tape:
+                qml.RX(x[0], wires=[0])
+                qml.RY(x[1], wires=[1])
+
+                for i in range(2, num_params):
+                    qml.RZ(x[i], wires=[i % 2])
+
+                qml.CNOT(wires=[0, 1])
+                qml.var(qml.PauliZ(0) @ qml.PauliX(1))
+
+            return execute([tape], dev, gradient_fn=param_shift, cache=cache, interface="torch")[0]
+
+        # No caching: number of executions is not ideal
+        hess1 = torch.autograd.functional.hessian(lambda x: cost(x, cache=None), params)
+
+        if num_params == 2:
+            # compare to theoretical result
+            x, y, *_ = params.detach()
+            expected = torch.tensor(
+                [
+                    [2 * np.cos(2 * x) * np.sin(y) ** 2, np.sin(2 * x) * np.sin(2 * y)],
+                    [np.sin(2 * x) * np.sin(2 * y), -2 * np.cos(x) ** 2 * np.cos(2 * y)],
+                ]
+            )
+            assert np.allclose(expected, hess1, atol=tol, rtol=0)
+
+        expected_runs = 1  # forward pass
+        expected_runs += 2 * N  # Jacobian
+        expected_runs += 4 * N + 1  # Hessian diagonal
+        expected_runs += 4 * N ** 2  # Hessian off-diagonal
+        assert dev.num_executions == expected_runs
+
+        # Use caching: number of executions is ideal
+        dev._num_executions = 0
+        hess2 = torch.autograd.functional.hessian(lambda x: cost(x, cache=True), params)
+        assert np.allclose(hess1, hess2, atol=tol, rtol=0)
+
+        expected_runs_ideal = 1  # forward pass
+        expected_runs_ideal += 2 * N  # Jacobian
+        expected_runs_ideal += 2 * N + 1  # Hessian diagonal
+        expected_runs_ideal += 4 * N * (N - 1) // 2  # Hessian off-diagonal
+        assert dev.num_executions == expected_runs_ideal
+        assert expected_runs_ideal < expected_runs
+
+    def test_caching_adjoint_backward(self):
+        """Test that caching reduces the number of adjoint evaluations
+        when mode=backward"""
+        dev = qml.device("default.qubit", wires=2)
+        params = torch.tensor([0.1, 0.2, 0.3])
+
+        def cost(a, cache):
+            with qml.tape.JacobianTape() as tape:
+                qml.RY(a[0], wires=0)
+                qml.RX(a[1], wires=0)
+                qml.RY(a[2], wires=0)
+                qml.expval(qml.PauliZ(0))
+                qml.expval(qml.PauliZ(1))
+
+            return execute(
+                [tape],
+                dev,
+                gradient_fn="device",
+                cache=cache,
+                mode="backward",
+                gradient_kwargs={"method": "adjoint_jacobian"},
+                interface="torch",
+            )[0]
+
+        # Without caching, 3 evaluations are required.
+        # 1 for the forward pass, and one per output dimension
+        # on the backward pass.
+        torch.autograd.functional.jacobian(lambda x: cost(x, cache=None), params)

```

Comment:
Yes 😆 

@trbromley brought up the same thing when reviewing #1508. It is my fault - I ended up supporting both, because I was getting confused while writing the tests and kept using `None`/`False`.

I figured if I was making this mistake repeatedly, other devs might as well lol

Reply:
Oh, I see! :smile: It might be worth allowing booleans only (in a separate PR) and going only with `cache=False`, just to remove any added ambiguity in the long run.